File: ProjectModuleThree/MySheetChatDa/app/src/main/java/com/example/mysheetchatda/Signup.java
package com.example.mysheetchatda;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.app.ProgressDialog;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import com.example.mysheetchatda.Models.User;
import com.example.mysheetchatda.databinding.ActivitySignupBinding;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.database.FirebaseDatabase;

/*
- Name: Adrianus Jonathan Engelbracht
-  Matriculation number: 1151826
- Date: 21.01.2024
*/
public class Signup extends AppCompatActivity {

    ActivitySignupBinding binding;
    private FirebaseAuth mAuth;
    FirebaseDatabase database;
    ProgressDialog progressDialog;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivitySignupBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        mAuth = FirebaseAuth.getInstance();
        database = FirebaseDatabase.getInstance();

        progressDialog = new ProgressDialog(Signup.this);
        progressDialog.setTitle("Creating Account");
        progressDialog.setMessage("Your account is being created");

        binding.btnSignUp.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v){
                if(!binding.txtUsername.getText().toString().isEmpty()
                        && !binding.txtEmail.getText().toString().isEmpty()
                        && !binding.txtPassword.getText().toString().isEmpty()
                ){
                    progressDialog.show();
                    mAuth.createUserWithEmailAndPassword(binding.txtEmail.getText().toString(), binding.txtPassword.getText().toString())
                            .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                                @Override
                                public void onComplete(@NonNull Task<AuthResult> task) {
                                    progressDialog.dismiss();
                                    if(task.isSuccessful()){
                                        User user = new User(binding.txtUsername.getText().toString(), binding.txtEmail.getText().toString(), binding.txtPassword.getText().toString());
                                        //Toast.makeText(signup.this, "u: " + user.getUserName() + "e: " + user.geteMail() + "p: " + user.getPassword(), Toast.LENGTH_SHORT).show();
                                        String id = task.getResult().getUser().getUid();
                                        //Toast.makeText(signup.this, "Id" + id, Toast.LENGTH_SHORT).show();
                                        database.getReference().child("User").child(id).setValue(user);
                                        Toast.makeText(Signup.this, "Success!", Toast.LENGTH_SHORT).show();

                                        if(mAuth.getCurrentUser() != null){
                                            Intent intent = new Intent(Signup.this, MainActivity.class);
                                            startActivity(intent);
                                        }
                                    }
                                    else{
                                        Toast.makeText(Signup.this, task.getException().toString(), Toast.LENGTH_SHORT).show();
                                    }
                                }
                            });

                }else{
                    Toast.makeText(Signup.this, "Else Ente", Toast.LENGTH_SHORT).show();
                }
            }
        });

        binding.txtAlreadyAccount.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(Signup.this, Signin.class);
                startActivity(intent);
            }
        });

    }
}

File: MyChat/app/src/main/java/com/example/mychat/Signup.kt
package com.example.mychat

import android.annotation.SuppressLint
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.provider.ContactsContract.CommonDataKinds.Email
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.firestore.auth.User
import java.net.PasswordAuthentication

class Signup : AppCompatActivity() {

    private lateinit var edtName: EditText
    private lateinit var edtEmail: EditText
    private lateinit var edtPasswordAuthentication: EditText
    private lateinit var btnSignup: Button
    private lateinit var mAuth: FirebaseAuth
    private lateinit var mDbRef: DatabaseReference

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_signup)

        mAuth = FirebaseAuth.getInstance()

        edtName = findViewById(R.id.edt_name)
        edtEmail = findViewById(R.id.edt_email)
        edtPasswordAuthentication = findViewById(R.id.edt_password)
        btnSignup = findViewById(R.id.btnSignup)

        btnSignup.setOnClickListener {
            val name = edtName.text.toString()
            val email = edtEmail.text.toString()
            val passwordAuthentication = edtPasswordAuthentication.text.toString()

            signUp(name, email, passwordAuthentication)
        }
    }

    private fun signUp(name: String, email: String, passwordAuthentication: String) {
        //logic of creating user
        mAuth.createUserWithEmailAndPassword(email, passwordAuthentication)
            .addOnCompleteListener(this) { task ->
                if (task.isSuccessful) {
                    //code for jumping to home
                    addUserToDatabase(name, email, mAuth.currentUser?.uid!!)
                    val intent = Intent(this@Signup, MainActivity::class.java)
                    finish()
                    startActivity(intent)


                } else {
                    Toast.makeText(this@Signup, "Some error occured", Toast.LENGTH_SHORT).show()

                }
            }
    }

    @SuppressLint("RestrictedApi")
    private fun addUserToDatabase(name: String, email: String, uid: String) {
        mDbRef = FirebaseDatabase.getInstance().getReference()

        mDbRef.child("user").child(uid).setValue(com.example.mychat.User(name,email,uid))
    }

}


